### PR TITLE
Add support for COPY command

### PR DESCRIPTION
**Pull request by dengliming**
Resolves [https://github.com/redis/jedis/issues/2471](https://github.com/redis/jedis/issues/2471)

### CODE REVIEW

File: src/main/java/redis/clients/jedis/params/CopyParams.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  private Long destinationDb;
+
+  private boolean replace;
+
+  public static CopyParams copyParams() {
+    return new CopyParams();
+  }
+
+  public CopyParams destinationDb(long destinationDb) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/params/CopyParams.java:22]: `db` can't be used in cluster mode. But `CopyParams` is directly used in cluster classes. This is contradictory.

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/main/java/redis/clients/jedis/params/CopyParams.java (hunk 1 above)
Comment (dengliming) [reply src/main/java/redis/clients/jedis/params/CopyParams.java]: em. Do you have any suggestions?

**Review by sazzad16 — CHANGES_REQUESTED** [review]
IMO, we can skip CopyParams for now.

**Comment by dengliming** [comment]
But I think `REPLACE` still needed.

File: src/main/java/redis/clients/jedis/commands/MultiKeyCommands.java [hunk 2]
```diff
@@ -21,6 +22,11 @@
 import java.util.Set;
 
 public interface MultiKeyCommands {
+
+  Long copy(String srcKey, String dstKey, CopyParams params);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/MultiKeyCommands.java:26]:
```suggestion
  boolean copy(String srcKey, String dstKey, Integer db, boolean replace);
```
WDYT?

Comment (dengliming) [reply src/main/java/redis/clients/jedis/commands/MultiKeyCommands.java]: I agree with returning `boolean`.  but `db` is not required. people need to pass null If they don't want it(That's why i introduce the param class). anyway They all fine to me. : )

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/commands/MultiKeyCommands.java:26]: If they don't want it, they can use the other method.

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/commands/MultiKeyCommands.java:26]:
```suggestion
  boolean copy(String srcKey, String dstKey, int db, boolean replace);
```

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/commands/MultiKeyCommands.java:26]:
```suggestion
  boolean copy(String srcKey, String dstKey, int db, boolean replace);
```
Sorry, my suggestion could be even better.

Comment (dengliming) [reply src/main/java/redis/clients/jedis/commands/MultiKeyCommands.java]:
```
boolean copy(String srcKey, String dstKey, int db, boolean replace);// not for cluster
boolean copy(String srcKey, String dstKey, boolean replace);
```
So, all we need to do is add the above two methods. Is that right?

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/commands/MultiKeyCommands.java:26]: yes

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**